vfs/vfscache/item.go

```diff
@@ -414,6 +428,12 @@ func (item *Item) _dirty() {
 
 // Dirty marks the item as changed and needing writeback
 func (item *Item) Dirty() {
+	item.preAccess()
```

Comment:
Rather than splitting the function, I think you could be using

```go
item.preAccess()
defer item.postAccess()
```

Which means `item.postAccess()` will be executed regardless of exit point (and also if a panic happens)

This is a much more Go way of doing it I think.

You might want to do this for all the functions as it does have the extra (very minor in this case) advantage of being run during a panic.

Note that in go1.15 `defer` calls are almost free - the compiler does static analysis and effectively turns them into old C-style `goto out;` error handling behind the scene.

Reply:
Thanks for the comments.  They make sense.  I will rework this commit accordingly.

Reply:
This commit has been redone according to the suggestion.